report/src/sections/07-future-works.tex

```diff
@@ -7,8 +7,18 @@ \section{Future Works}\label{sec:future-works}
 The extension of the abstract interpretation to support floating point numbers would allow for a more precise analysis of the database queries.
 One of the obstacles to this extension is the fact that floating point numbers are represented in a different ways in different programming languages, which makes it difficult to create a general abstract interpretation for floating point numbers.
 
-Another area that could be further developed is the precision of the semantic in som cases.
-The current set of semantics are very similar to the ones fund in the original paper by Halder and Cortesi~\cite{halder_abstract_2012}, and they are not very precise.
-This could then lead to a more precise analysis of the database area and more accurate results when judging the abstract interpretation of the database.
+Another area that could be further developed is the rigor of the soundness proof.
+Since you can keep reducing the area represented by the abstract interpretation, given more time it would be natural to assume that the soundness proof could be made more concise to further narrow down the area represented by the abstract interpretation.
+This is a very time-consuming process, but it would be interesting to see how far you could take it.
+Though the more you try to shrink the area, the more edge cases appear which need to be handled.
```

Comment:
This is still not related to the proof, right now the proof is more of an argument of soundsness. If the proof was rigorous we would be confident that our analysis is an overapproximation of the state-space.
The proof just shows that the 'area' of the analysis is bigger than the area of actual state-space. The analysis is what determines the precision not the proof.